test/core/models/ocelot/stanzas/SequenceStanzaSpec.scala

```diff
@@ -86,6 +86,28 @@ class SequenceStanzaSpec extends BaseSpec {
       sequence.validInput("2,3") shouldBe Some("2,3")
       sequence.validInput("0") shouldBe Some("0")
     }
+
+    "assign Nil to labels property when no label is used" in {
+      val seq = Sequence(expectedStanza.copy(label = None), Phrase("",""), Seq(Phrase("",""),Phrase("",""),Phrase("",""),Phrase("","")))
+      seq.labels shouldBe Nil
+    }
+
+    "Evaluate invalid input to error return" in {
+      val labels = LabelCache()
+      val blankPage: Page = Page("any", "/url", Seq.empty, Nil)
+      val sequence = Sequence(expectedStanza, Phrase("",""), Seq(Phrase("",""),Phrase("",""),Phrase("",""),Phrase("","")))
+      val noopReturn: (Option[String], Labels) = (None, labels)
+      sequence.eval("hello", blankPage, labels) shouldBe noopReturn
+    }
+
+    "Ignore input indexes which dont exist in the options list" in {
+      val labels = LabelCache()
+      val blankPage: Page = Page("any", "/url", Seq.empty, Nil)
+      val sequence = Sequence(expectedStanza, Phrase("",""), Seq(Phrase("",""),Phrase("",""),Phrase("",""),Phrase("","")))
+      val noopReturn: (Option[String], Labels) = (None, labels)
+      sequence.eval("24", blankPage, labels) shouldBe noopReturn
+      labels.stackList shouldBe Nil
+    }
```

Comment:
I think we could do with a positive test here to check that the label stack is created correctly.

Reply:
will add